### PR TITLE
tkt-65943: Allow non-word characters (like .) in pool/dataset/zvol names in API 1.0

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by sonicaj** [comment]
`The pool name must begin with a letter, and can only contain alphanumeric characters as well as underscore ("_"), dash ("-"), and period (".")`

File: gui/api/resources.py [hunk 1]
```diff
@@ -738,109 +738,109 @@ def obj_get(self, bundle, **kwargs):
     def prepend_urls(self):
         return [
             url(
-                r"^(?P<resource_name>%s)/(?P<pk>\w[\w/-]*)/datasets%s$" % (
```
Comment (sonicaj) [inline gui/api/resources.py]: I am not sure why we allowed forward slashes in the regex before. Would you know if we had a reason for it before and if it's okay to keep on allowing it ?

Comment (themylogin) [reply gui/api/resources.py]: It has always been like this, I don't know for which reason, but for sake of not breaking anything let's keep it.

Comment (sonicaj) [reply gui/api/resources.py]: Thank you for clarifying :)

**Pull request merged**